BelfDev/homely
# Integrate the Homely API with Postgres Database

This PR integrates the Homely API with a PostgreSQL database, enabling persistent data storage for the application. Key changes include setting up the database configuration, adding migrations for the initial schema, and ensuring that the API interacts smoothly with the Postgres database.

## Key Changes

### Database Setup:
- Configured PostgreSQL as the database for the API using SQLAlchemy and Flask-SQLAlchemy.
- Updated the environment configuration to include necessary database variables (POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_DB, DATABASE_URL).

### Docker Integration:
- Updated the docker-compose.dev.yml to include a Postgres service, ensuring the database is running alongside the API.
- Modified the web service in the Docker Compose setup to establish a connection to the Postgres container.

### Docker Compose Configuration:
- Updated the docker-compose.dev.yml to include a db service running PostgreSQL and a web service for the API.
- The web service builds the Flask app and connects to the database.
- Exposed ports for both services:
  - API: Exposed on port 5050, mapped to port 5050 inside the container.
  - Postgres: Exposed on port 5432 for database access.

### Flask-Migrate Integration:
- Added Flask-Migrate to handle database migrations.
- Created an initial migration script with flask db migrate.
- Applied migrations using flask db upgrade to set up the database schema.

### Makefile Commands:
- Added useful Makefile commands for database operations, including migration and upgrade commands, and a way to easily access the Postgres shell inside the Docker container.

### How to Test
1. Pull the latest changes from this branch.
2. Ensure you have the .env file with valid Postgres configuration.
3. Run make start to build and start the API and database containers.
4. Verify that the API is accessible at http://localhost:5000/api/v1/health.
5. Test database operations such as user creation and querying to ensure proper integration.
6. Use make db-migrate and make db-upgrade to test migrations and schema updates.

## Notes
- This PR lays the groundwork for future features that require persistent data.
- Make sure to check the .env file for the correct database credentials before starting the application.
